lib/floe/validation_mixin.rb

```diff
@@ -23,7 +23,7 @@ def runtime_field_error!(field_name, field_value, comment, floe_error: "States.R
     end
 
     def workflow_state?(field_value, workflow)
-      workflow.payload["States"] ? workflow.payload["States"].include?(field_value) : true
+      workflow.workflow_state?(field_value, workflow)
```

Comment:
What if instead of `workflow.payload["States"]` we use `workflow.states_by_name.key?(field_value)`

Reply:
Do we need to pass in `workflow` as an argument if we're explicitly calling this on the `workflow` instance?  Seems redundant.

Reply:
It seemed easier to throw up a PR to look at the diff, what do you think about https://github.com/ManageIQ/floe/pull/326/files ?